TestServerPathEncodingIssues/Windows_multilingual_404 fails on Debian buildd armel and s390x

It looks like TestServerPathEncodingIssues/Windows_multilingual_404 in v0.104.0 is another "flaky" test (see https://github.com/gohugoio/hugo/commit/08f0984f918bb97d35c1765d3b94b1a9a1a3b9cd) , failing on the first try on [armel](https://buildd.debian.org/status/logs.php?pkg=hugo&arch=armel) and [s390x](https://buildd.debian.org/status/logs.php?pkg=hugo&arch=s390x) architectures on Debian buildd machines:

```console
=== RUN   TestServerPathEncodingIssues/Windows_multilingual_404
    server_test.go:343: 
        error:
          got non-nil error
        got:
          e`Get "http://localhost:41561/en/this/does/not/exist": dial tcp [::1]:41561: connect: connection refused`
        stack:
          /<<PKGBUILDDIR>>/_build/src/github.com/gohugoio/hugo/commands/server_test.go:343
            c.Assert(err, qt.IsNil)
          /<<PKGBUILDDIR>>/_build/src/github.com/gohugoio/hugo/commands/server_test.go:353
            func() {
                resp, err := http.Get(fmt.Sprintf("http://localhost:%d/%s", port, path))
                c.Assert(err, qt.IsNil)
                pr := pathResult{
                    statusCode: resp.StatusCode,
                }
            
                if err == nil {
                    defer resp.Body.Close()
                    pr.body = helpers.ReaderToString(resp.Body)
                }
                result.pathsResults[path] = pr
            }()
          /<<PKGBUILDDIR>>/_build/src/github.com/gohugoio/hugo/commands/server_test.go:124
            r := runServerTest(c,
                serverTestOptions{
                    config:     config,
                    pathsToGet: []string{"en/this/does/not/exist", "es/this/does/not/exist"},
                },
            )
        
--- FAIL: TestServerPathEncodingIssues (1.07s)
    --- PASS: TestServerPathEncodingIssues/Unicode_paths (1.06s)
    --- FAIL: TestServerPathEncodingIssues/Windows_multilingual_404 (0.01s)
```

I tried rebuilding (or "giveback" in Debian-speak), but surprisingly to me, the test failed consistently on armel and s390x (and ppc64)

Don't know why yet; I'll try again on Debian porterbox machines for armel and s390x and see if I could reproduce the error.

<!-- Please answer these questions before submitting your issue. Thanks! -->

### What version of Hugo are you using (`hugo version`)?

```
$ hugo version
hugo v0.104.0+extended linux/s390x BuildDate=2022-09-25T07:58:54Z VendorInfo=debian:0.104.0-1
```

### Does this issue reproduce with the latest release?

Yes, only tested with v0.104.0.

--- a/commands/server_test.go
+++ b/commands/server_test.go
@@ -25,7 +25,6 @@ import (
 
 	"github.com/gohugoio/hugo/config"
 	"github.com/gohugoio/hugo/helpers"
-	"github.com/gohugoio/hugo/htesting"
 	"golang.org/x/net/context"
 	"golang.org/x/sync/errgroup"
 
@@ -69,20 +68,14 @@ func TestServer404(t *testing.T) {
 }
 
 func TestServerPathEncodingIssues(t *testing.T) {
-	if htesting.IsGitHubAction() {
-		// This test is flaky on CI for some reason. Run it on Windows only for now.
-		// TODO(bep)
-		if runtime.GOOS != "windows" {
-			t.Skip("Skipping test on CI")
-		}
-	}
 	c := qt.New(t)
 
 	// Issue 10287
 	c.Run("Unicode paths", func(c *qt.C) {
 		r := runServerTest(c,
 			serverTestOptions{
-				pathsToGet: []string{"hügö/"},
+				pathsToGet:  []string{"hügö/"},
+				getNumHomes: 1,
 			},
 		)
 
@@ -123,8 +116,9 @@ status = 404
 `
 		r := runServerTest(c,
 			serverTestOptions{
-				config:     config,
-				pathsToGet: []string{"en/this/does/not/exist", "es/this/does/not/exist"},
+				config:      config,
+				pathsToGet:  []string{"en/this/does/not/exist", "es/this/does/not/exist"},
+				getNumHomes: 1,
 			},
 		)
 

